### DISCUSSION TITLE
ControlledTree

### DISCUSSION THREAD

**Discussion by paprophet** [iTwin/itwinjs-core#2328] [Q&A] [answered]
now use the ControlledTree,but  i have 2 questions：
1. how support the contextmenu in the ControlledTree ,(show the menu  when  right the node or the tree  )  ?
2. open a define the node when loaded  the tree data. (  the tree expend to the element in   the element tree   when i click the element in the model, )

**Comment by roluk** [comment, marked as answer]
Hello @paprophet,

1. You will need to create a custom React component that implements your desired interactions, and pass it through `nodeRenderer` prop to `TreeRenderer`, as demonstrated in [Custom Table Node Tree sample](https://www.itwinjs.org/sample-showcase/?group=UI+Trees&sample=custom-table-node-tree-sample).

2. You can expand a node programmatically by calling TreeModelSource's `modifyModel` method:

    ```ts
    function expandFirstNode(modelSource: TreeModelSource): void {
      modelSource.modifyModel((treeModel) => {
        const firstNode = treeModel.getNode(undefined, 0);
        if (isTreeModelNode(firstNode)) {
          firstNode.isExpanded = true;
        }
      });
    }
    ```

**Comment by paprophet** [comment]
thank you @roluk
